pkg/deploy/lattice/service_network_manager.go

```diff
@@ -274,22 +285,78 @@ func (m *defaultServiceNetworkManager) isServiceNetworkAssociatedWithVPC(ctx con
 				switch associationStatus {
 				case vpclattice.ServiceNetworkVpcAssociationStatusActive:
 					glog.V(6).Infoln("Mesh and Vpc association is active.")
-					return true, r.Id, resp, nil
+					return true, r, resp, nil
 				case vpclattice.ServiceNetworkVpcAssociationStatusCreateFailed:
 					glog.V(6).Infoln("Mesh and Vpc association does not exists, start creating service_network and vpc association")
-					return false, r.Id, resp, nil
+					return false, r, resp, nil
 				case vpclattice.ServiceNetworkVpcAssociationStatusDeleteFailed:
 					glog.V(6).Infoln("Mesh and Vpc association failed to delete")
-					return true, r.Id, resp, nil
+					return true, r, resp, nil
 				case vpclattice.ServiceNetworkVpcAssociationStatusDeleteInProgress:
 					glog.V(6).Infoln("ServiceNetwork and Vpc association is being deleted, please retry later")
-					return true, r.Id, resp, errors.New(LATTICE_RETRY)
+					return true, r, resp, errors.New(LATTICE_RETRY)
 				case vpclattice.ServiceNetworkVpcAssociationStatusCreateInProgress:
 					glog.V(6).Infoln("ServiceNetwork and Vpc association is being created, please retry later")
-					return true, r.Id, resp, errors.New(LATTICE_RETRY)
+					return true, r, resp, errors.New(LATTICE_RETRY)
 				}
 			}
 		}
 	}
 	return false, nil, resp, err
 }
+
+func (m *defaultServiceNetworkManager) UpdateServiceNetworkVpcAssociationIfNeeded(ctx context.Context, snId string, snArn string, desiredSN *latticemodel.ServiceNetwork, existingSnvaId *string) (latticemodel.ServiceNetworkStatus, error) {
+	retrievedSnva, err := m.cloud.Lattice().GetServiceNetworkVpcAssociationWithContext(ctx, &vpclattice.GetServiceNetworkVpcAssociationInput{
+		ServiceNetworkVpcAssociationIdentifier: existingSnvaId,
+	})
+	if err != nil {
+		return latticemodel.ServiceNetworkStatus{}, err
+	}
+
+	if len(desiredSN.Spec.SecurityGroupIds) == 0 && retrievedSnva.SecurityGroupIds != nil && len(retrievedSnva.SecurityGroupIds) > 0 {
+		// VPC latiice API have a limitation that the user can't remove all security groups if current snva already has security groups.
+		// If that case happen, the controller will not delete snva. User should manually delete it.
+		//https://docs.aws.amazon.com/vpc-lattice/latest/ug/security-groups.html#security-groups-rules
```

Comment:
this use-case should print error log, so operator can read log and do action

Reply:
Will change to error log